docs/criteria/accelerator-stages.md

```diff
@@ -9,51 +9,41 @@ Current supported data accelerators (DuckDB, Postgres, SQlite, Arrow) release st
 ### Feature set complete
 
 - [ ] Streaming when accelerating/loading
-- [ ] Index
+- [ ] Streaming when querying
+- [ ] Supports primary keys and indexes
+- [ ] Supports federation in a single dataset
+- [ ] Supports push-down federation across multiple datasets within the same accelerator
 - [ ] On conflict behaviors - Drop/Upsert
-- [ ] File mode for SQLite and DuckDB
+- [ ] File mode for supported accelerator engines, i.e. SQLite, DuckDB, etc.
 - [ ] Common/core arrow data type supports: Null, all Int types, all Float types, Time32, Time64, Timestamp (with/without TZ), Date32, Date64, Duration, Interval, Binary/LargeBinary/FixedSizeBinary, Utf8/LargeUtf8, List/FixedSizeList/LargeList, Struct, Decimal128/Decimal256
 - [ ] No known major bugs/issues
 
 ### Test Coverage
 
 - [ ] End-to-end test to cover accelerating TPCH-SF1 data from S3 and benchmarking TPCH queries (official and simple).
-  - file mode if applicable
-  - memory mode if applicable
+  - [ ] File mode
+  - [ ] Embedded memory mode
 - [ ] Integration tests to cover accelerating data from S3 parquet, MySQL, Postgres with arrow types: Null, all Int types, all Float types, Time32, Time64, Timestamp (with/without TZ), Date32, Date64, Duration, Interval, Binary/LargeBinary/FixedSizeBinary, Utf8/LargeUtf8, List/FixedSizeList/LargeList, Struct, Decimal128/Decimal256
 - [ ] Integration tests to cover "On Conflict" behaviors.
-- [ ] A reproducible test script that can load a variety of datasets into accelerators in both file/memory mode.
-  - Can load below datasets
-    - [ ] TPCH-SF10
-    - [ ] TPCH-SF100
-    - [ ] TPCDS-SF10
-    - [ ] TPCDS-SF100
-    - [ ] db-benchmark join-datagen 50G (from datafusion benchmark)
-    - [ ] clickbench_1 (from datafusion benchmark)
-    - [ ] Semi-structure/Blob based data with row size 1M, 10M, 100M
-  - Can run smoke test on accelerating
-    - [ ] Load SF10 into memory
-    - [ ] Load SF100 into file
-  - Can run smoke test on querying on accelerated data
-    - [ ] Query on SF10 dataset with memory mode
-    - [ ] Query on SF100 dataset with file mode
+- [ ] A reproducible test script that can load a variety of datasets into accelerators in both file/embedded memory mode.
+  - Loads and runs these benchmarks with no errors:
+    - [ ] TPCH-SF10 (embedded memory/file)
+    - [ ] TPCH-SF100 (file)
+    - [ ] TPCDS-SF10 (embedded memory/file)
+    - [ ] TPCDS-SF100 (file)
+    - [ ] Clickbench
```

Comment:
```suggestion
  - Loads and runs benchmarks with no errors:
    - [ ] TPCH-SF10 (embedded memory/file)
    - [ ] TPCH-SF100 (file)
    - [ ] TPCDS-SF10 (embedded memory/file)
    - [ ] TPCDS-SF100 (file)
    - [ ] Clickbench
```

Reply:
We should add `Spicebench` for the specific edge cases we need to support, including DECIMAL256, etc.